src/crimson/os/seastore/transaction.h

```diff
@@ -178,9 +192,11 @@ class Transaction {
 
   void add_mutated_extent(CachedExtentRef ref) {
     ceph_assert(!is_weak());
-    assert(read_set.count(ref->prior_instance->get_paddr()));
+    assert(ref->is_exist_mutation_pending() ||
+	   read_set.count(ref->prior_instance->get_paddr()));
     mutated_block_list.push_back(ref);
-    write_set.insert(*ref);
+    if (!ref->is_exist_mutation_pending())
+      write_set.insert(*ref);
```

Comment:
Always use braces for if statements (check the rest of the PR as well)

```
    if (!ref->is_exist_mutation_pending()) {
      write_set.insert(*ref);
    }
```

Also, can we assert that *ref must already be in the write_set?

```
    if (!ref->is_exist_mutation_pending()) {
      write_set.insert(*ref);
    } else {
      assert(write_set.count(*ref));
    }
```